src/settings_internal.js

```diff
@@ -273,3 +273,7 @@ var BULK_MEMORY = false;
 var MINIFY_WHITESPACE = true;
 
 var ASYNCIFY_IMPORTS_EXCEPT_JS_LIBS = [];
+
+// Specify which ports to use. This is set via the --use-port command line
+// option
+var PORTS = [];
```

Comment:
This this is currently used used in the `ports/__init__.py`, maybe lets just make it a global varaible there.

Perhaps a set too?  e.g. `needed = set()` at the of `__init__.py`?

Reply:
great idea that I implemented